newswires/client/src/context/SearchContext.tsx

```diff
@@ -167,15 +180,34 @@ export function SearchContextProvider({ children }: PropsWithChildren) {
 				});
 		}
 
-		if (state.status === 'success' || state.status === 'offline') {
+		if (state.status === 'loading-more') {
+			fetchResults(currentConfig.query, {
```

Comment:
I wonder whether it would be more straightforward to put the `fetch().then().catch()` into the body of the `loadMoreResults` function? It will hopefully keep the number of branches in the `status` state machine simpler, and as far as I can tell it's only the load more button itself that needs to know about whether the fetch is underway, so we could perhaps have 'loading-more' as a piece of more local state. What do you think?